src/udev/udev-builtin-net_id.c

```diff
@@ -794,6 +797,44 @@ static int names_mac(sd_device *dev, struct netnames *names) {
         return 0;
 }
 
+static int names_netdevsim(sd_device *dev, struct netnames *names) {
+        sd_device *netdevsimdev;
+        const char *sysname;
+        unsigned int addr;
+        const char *port_name = NULL;
+        size_t l;
+        char *s;
+        int r;
+
+        assert(dev);
+        assert(names);
+
+        r = sd_device_get_parent_with_subsystem_devtype(dev, "netdevsim", NULL, &netdevsimdev);
+        if (r < 0)
+                return r;
+
+        r = sd_device_get_sysname(netdevsimdev, &sysname);
+        if (r < 0)
+                return r;
+
+        if (sscanf(sysname, "netdevsim%u", &addr) != 1)
+                return -EINVAL;
+
+        (void) sd_device_get_sysattr_value(dev, "phys_port_name", &port_name);
```

Comment:
The current implementation does not allow devices without `port_name`. So, please return earlier than writing the address in `names->netdevsim_path`. That is,
```
r = sd_device_get_sysattr_value(dev, "phys_port_name", &port_name);
if (r < 0)
        return r;
```